packages/vitest/src/node/plugins/index.ts

```diff
@@ -50,18 +62,15 @@ export async function VitestPlugin(options: UserConfig = {}, ctx = new Vitest())
       },
       async configResolved(viteConfig) {
         const viteConfigTest = (viteConfig.test as any) || {}
-        if (viteConfigTest.watch === false)
-          viteConfigTest.run = true
+        if (viteConfigTest.watch === false) viteConfigTest.run = true
 
         // viteConfig.test is final now, merge it for real
-        options = deepMerge(
-          {},
-          configDefaults,
-          viteConfigTest,
-          options,
-        )
+        options = deepMerge({}, configDefaults, viteConfigTest, options)
         options.api = resolveApiConfig(options)
-        options.watch = options.watch && !options.run
+        if (isWatchArg())
```

Comment:
Having the plugin coupled with CLI argv is probably not a good idea. I will take another try